clients/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumerImpl.java

```diff
@@ -182,30 +231,110 @@ public Collection<PubSubMessage<K, ChangeEvent<V>, Long>> poll(long timeoutInMs)
     return pubSubMessages;
   }
 
+  protected <T> T bufferAndAssembleRecordChangeEvent(
+      PubSubTopicPartition pubSubTopicPartition,
+      int schemaId,
+      byte[] keyBytes,
+      byte[] valueBytes,
+      long recordOffset,
+      AbstractAvroChunkingAdapter<T> chunkingAdapter,
+      Lazy<RecordDeserializer<T>> recordDeserializer,
+      int readerSchemaId,
+      ReadOnlySchemaRepository schemaRepository) {
+    T recordChangeEvent = null;
+    if (storeChunkingEnabled) {
+      if (!inMemoryStorageEngine.containsPartition(pubSubTopicPartition.getPartitionNumber())) {
+        inMemoryStorageEngine.addStoragePartition(pubSubTopicPartition.getPartitionNumber());
+      }
+      if (schemaId == AvroProtocolDefinition.CHUNK.getCurrentProtocolVersion()) {
```

Comment:
I'd like to confirm that every time this client start will it always consume from the beginning of the topic or do we checkpoint any offset somewhere? If we do checkpointing maybe we need make sure only when we fully construct a record and we update the checkpoint offset. I don't 100% sure about that part of impl so checking here.

Reply:
Checkpointing doesn't work today, but need to add it in the coming weeks.  My order of work prioritization is thus: Make the client library available so folks can start developing, then add the seek functionality.  I'm waiting to do it because the partner team hasn't made it entirely clear what requirements they have for the checkpoint.  They need this checkpoint to work like an SCN on espresso/brooklin, but it's not been explicitly stated yet what qualities of that they specifically rely on.

So I'm holding off for now, but it's coming soon!